sentry_sdk/integrations/flask.py

```diff
@@ -73,30 +78,77 @@ def _before_request(*args, **kwargs):
                 scope.transaction = request.url_rule.endpoint
 
             try:
-                scope.request = _get_request_info()
+                _set_request_info(scope)
             except Exception:
                 get_current_hub().capture_internal_exception()
 
             try:
-                scope.user = _get_user_info()
+                _set_user_info(scope)
             except Exception:
-                raise
                 get_current_hub().capture_internal_exception()
     except Exception:
         get_current_hub().capture_internal_exception()
 
-def _get_request_info():
-    return {
-        'url': '%s://%s%s' % (request.scheme, request.host, request.path),
+def _set_request_info(scope):
+    request_info = {
+        'url': '%s://%s%s' % (
+            request.scheme,
+            request.host,
+            request.path
+        ),
         'query_string': request.query_string,
         'method': request.method,
-        'data': request.get_data(cache=True, as_text=True, parse_form_data=True),
         'headers': dict(request.headers),
-        'env': get_environ(request.environ)
+        'env': dict(get_environ(request.environ))
     }
 
-
-def _get_user_info():
+    scope.request = request_info
+    # if this crashes we at least have the rest of the request already set
+    _set_request_body(request_info, scope)
+
+
+def _set_request_body(request_info, scope):
```

Comment:
I will have a serious problem with code reuse for the django integration here, that's why I am considering doing it on the wsgi level. But alas Django allegedly sucks for adding custom wsgi middleware anyway, so I'm not sure